src/neuroglancer/datasource/deepzoom/frontend.ts

```diff
@@ -0,0 +1,304 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {makeDataBoundsBoundingBoxAnnotationSet} from 'neuroglancer/annotation';
+import {ChunkManager, WithParameters} from 'neuroglancer/chunk_manager/frontend';
+import {BoundingBox, CoordinateSpace, makeCoordinateSpace, makeIdentityTransform, makeIdentityTransformedBoundingBox} from 'neuroglancer/coordinate_transform';
+import {WithCredentialsProvider} from 'neuroglancer/credentials_provider/chunk_source_frontend';
+import {CompleteUrlOptions, ConvertLegacyUrlOptions, DataSource, DataSourceProvider, DataSubsourceEntry, GetDataSourceOptions, NormalizeUrlOptions} from 'neuroglancer/datasource';
+import {ImageTileEncoding, ImageTileSourceParameters} from 'neuroglancer/datasource/deepzoom/base';
+import {responseText} from 'neuroglancer/datasource/dvid/api';
+import {parseProviderUrl, resolvePath, unparseProviderUrl} from 'neuroglancer/datasource/precomputed/frontend';
+import {SliceViewSingleResolutionSource} from 'neuroglancer/sliceview/frontend';
+import {makeDefaultVolumeChunkSpecifications, VolumeSourceOptions, VolumeType} from 'neuroglancer/sliceview/volume/base';
+import {MultiscaleVolumeChunkSource, VolumeChunkSource} from 'neuroglancer/sliceview/volume/frontend';
+import {transposeNestedArrays} from 'neuroglancer/util/array';
+import {DataType} from 'neuroglancer/util/data_type';
+import {completeHttpPath} from 'neuroglancer/util/http_path_completion';
+import {parseArray, parseFixedLengthArray, verifyEnumString, verifyFinitePositiveFloat, verifyInt, verifyObject, verifyObjectProperty, verifyOptionalObjectProperty, verifyPositiveInt, verifyString} from 'neuroglancer/util/json';
+import {getObjectId} from 'neuroglancer/util/object_id';
+import {cancellableFetchSpecialOk, parseSpecialUrl, SpecialProtocolCredentials, SpecialProtocolCredentialsProvider} from 'neuroglancer/util/special_protocol_request';
+
+/*export*/ class DeepzoomImageTileSource extends
+(WithParameters(WithCredentialsProvider<SpecialProtocolCredentials>()(VolumeChunkSource), ImageTileSourceParameters)) {}
+
+class ScaleInfo {
+  key: string;
+  encoding: ImageTileEncoding;
+  resolution: Float64Array;
+  voxelOffset: Float32Array;
+  size: Float32Array;
+  chunkSizes: Uint32Array[];
+  constructor(obj: any, numChannels: number) {
+    verifyObject(obj);
+    const rank = (numChannels === 1) ? 3 : 4;
+    const resolution = this.resolution = new Float64Array(rank);
+    const voxelOffset = this.voxelOffset = new Float32Array(rank);
+    const size = this.size = new Float32Array(rank);
+    if (rank === 4) {
+      resolution[3] = 1;
+      size[3] = numChannels;
+    }
+    verifyObjectProperty(
```

Comment:
This json parsing should be removed.

Reply:
Done in https://github.com/google/neuroglancer/commit/ceca06e04a75b5eb39eb435dbd3058decefb5d85